x-pack/plugins/watcher/server/routes/api/watches/register_list_route.ts

```diff
@@ -7,25 +7,12 @@
 
 import { IScopedClusterClient } from '@kbn/core/server';
 import { get } from 'lodash';
-import { fetchAllFromScroll } from '../../../lib/fetch_all_from_scroll';
-import { INDEX_NAMES, ES_SCROLL_SETTINGS } from '../../../../common/constants';
 import { RouteDependencies } from '../../../types';
 // @ts-ignore
 import { Watch } from '../../../models/watch';
 
 function fetchWatches(dataClient: IScopedClusterClient) {
-  return dataClient.asCurrentUser
-    .search(
-      {
-        index: INDEX_NAMES.WATCHES,
-        scroll: ES_SCROLL_SETTINGS.KEEPALIVE,
-        body: {
-          size: ES_SCROLL_SETTINGS.PAGE_SIZE,
-        },
-      },
-      { ignore: [404] }
-    )
-    .then((body) => fetchAllFromScroll(body, dataClient));
+  return dataClient.asCurrentUser.watcher.queryWatches();
```

Comment:
I think we should still specify the size param here as we were doing for the .search before, seems the new body default for the `.queryWatches` is 10 whereas before we were using 100

Reply:
I'm not sure if the size in the previous context should have the same value now. Here are my thoughts on this (correct me if I misunderstood something): before we made an initial Scroll search where we set the maximum number of hits returned with each batch of results to be 100. Then we passed this response to the [fetchAllFromScroll](https://github.com/elastic/kibana/blob/master/x-pack/plugins/watcher/server/lib/fetch_all_from_scroll/fetch_all_from_scroll.ts) function that recursively calls the Scroll API and each call returns the next batch of results until there are no more results left to return, ie the hits array is empty. In that context, the size was the maximum number of hits per batch of results.

With the Query watches API, the [docs](https://www.elastic.co/guide/en/elasticsearch/reference/8.6/watcher-api-query-watches.html) say that it "Retrieves all registered watches." so I thought that the size parameter is just used under the hood for performing multiple searches in a paginated manner. Or would it mean that we retrieve all registered watches ONLY in the range `from` to `from + size`? If that's the case then we would also need to make multiple calls to the Query watches API to get all hits. I'm not really sure... @martijnvg, I saw that you opened the [PR](https://github.com/elastic/elasticsearch/pull/64582) for adding the Query watches API, would you be able to provide more context on this?

Reply:
Hey @martijnvg, could you please help us with our question on the Query watches API or refer us to someone else who can help? Thanks in advance!

Reply:
> Or would it mean that we retrieve all registered watches ONLY in the range from to from + size?

Yes, that is correct. So multiple calls will need to be made in order to retrieve all watches. It is using the Search API under the hood and by default the `size` is `10`. You can also reach out to @elastic/es-data-management for questions about watcher and this api.

Reply:
Thanks @martijnvg! 

@sabarasaba in this case, I will implement a new function that will make multiple calls to the Query watches API until all hits are retrieved.